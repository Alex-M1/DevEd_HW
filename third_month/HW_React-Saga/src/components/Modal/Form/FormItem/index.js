import { connect } from 'react-redux';
import { setInputValue } from '../../../../store/actions';
import { iptValue } from '../../../../store/selectors';
import FormItem from './FormItem';

const mapStateToProps = (state, props) => ({
  value: iptValue(state, props.type),
});

const mapDispatchToProps = (dispatch, props) => ({
  onChange: (value) => dispatch(setInputValue(props.type, value)),
});

export default connect(mapStateToProps, mapDispatchToProps)(FormItem);
