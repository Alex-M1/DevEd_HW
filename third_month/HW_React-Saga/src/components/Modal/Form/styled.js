import styled from 'styled-components';

export const StForm = styled.div`
  display: grid;
  grid-template-rows: repeat(5,1fr);
  gap: 5px;
  padding: 10px 0;
  width: 100%;
`;

export const StFormItem = styled.div`
  display: grid;
  grid-template-columns: 30% 70%;
  gap: 5px;
  span{
    justify-self: flex-start;
  }
  input{
    padding: 3px 5px;
    border-radius: 4px;
    justify-self: end;
  }
`;
