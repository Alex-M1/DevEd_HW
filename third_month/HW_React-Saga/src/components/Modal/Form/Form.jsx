import React from 'react';
import FormItem from './FormItem';
import { StForm } from './styled';

function Form() {
  return (
    <StForm>
      <FormItem
        type="name"
        title="Имя"
        placeholder="Введите имя"
      />
      <FormItem
        type="age"
        title="Возраст"
        placeholder="Введите возраст"
      />
      <FormItem
        type="phone"
        title="Телефон"
        placeholder="Введите телефон"
      />
      <FormItem
        type="email"
        title="Email"
        placeholder="Введите email"
      />
      <FormItem
        type="company"
        title="Компания"
        placeholder="Название компании"
      />
    </StForm>
  );
}

export default Form;
