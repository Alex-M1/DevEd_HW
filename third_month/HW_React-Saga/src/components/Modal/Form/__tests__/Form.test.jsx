import React from 'react';
import Form from '../Form';
import { store } from './mockStore';

describe('Form', () => {
  it('Should match snapshot', () => {
    const component = global.shallowSmart(<Form />, store);
    expect(component.html()).toMatchSnapshot();
  });
  it('Should have StForm', () => {
    const component = global.mountSmart(<Form />, store);
    expect(component.find('styled__StForm')).toHaveLength(1);
  });
  it('Should have FormItem', () => {
    const component = global.mountSmart(<Form />, store);
    expect(component.find('FormItem')).toHaveLength(5);
  });
});
