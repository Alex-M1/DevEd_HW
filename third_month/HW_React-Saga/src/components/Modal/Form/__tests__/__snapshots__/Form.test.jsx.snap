// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form Should match snapshot 1`] = `"<div class=\\"styled__StForm-p2kz64-0 kNCWSz\\"><div class=\\"styled__StFormItem-p2kz64-1 kSxDuO\\"><span>Имя</span><input value=\\"\\" placeholder=\\"Введите имя\\"/></div><div class=\\"styled__StFormItem-p2kz64-1 kSxDuO\\"><span>Возраст</span><input value=\\"\\" placeholder=\\"Введите возраст\\"/></div><div class=\\"styled__StFormItem-p2kz64-1 kSxDuO\\"><span>Телефон</span><input value=\\"\\" placeholder=\\"Введите телефон\\"/></div><div class=\\"styled__StFormItem-p2kz64-1 kSxDuO\\"><span>Email</span><input value=\\"\\" placeholder=\\"Введите email\\"/></div><div class=\\"styled__StFormItem-p2kz64-1 kSxDuO\\"><span>Компания</span><input value=\\"\\" placeholder=\\"Название компании\\"/></div></div>"`;
